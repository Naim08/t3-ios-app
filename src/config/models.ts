export interface ModelOption {
  id: string;
  name: string;
  description: string;
  isPremium: boolean;
  icon?: string;
  provider?: 'openai' | 'anthropic' | 'google';
  category?: 'fast' | 'balanced' | 'advanced' | Array<'fast' | 'balanced' | 'advanced'>;
}

export const AI_MODELS: ModelOption[] = [
  {
    id: 'gpt-3.5-turbo',
    name: 'GPT-3.5 Turbo',
    description: 'Fast and efficient for everyday tasks',
    isPremium: false,
    provider: 'openai',
    category: 'fast',
  },
  {
    id: 'gpt-4o',
    name: 'GPT-4o',
    description: 'Latest model with enhanced reasoning',
    isPremium: true,
    provider: 'openai',
    category: 'advanced',
  },
  {
    id: 'claude-3-sonnet',
    name: 'Claude 3 Sonnet',
    description: 'Advanced reasoning and analysis',
    isPremium: true,
    provider: 'anthropic',
    category: 'advanced',
  },
  // Free Gemini models
  {
    id: 'gemini-pro',
    name: 'Gemini Pro',
    description: 'Google\'s advanced multimodal model',
    isPremium: false,
    provider: 'google',
    category: 'balanced',
  },
  {
    id: 'text-embedding-004',
    name: 'Text Embedding 004',
    description: 'Google\'s free text embedding model',
    isPremium: false,
    provider: 'google',
    category: 'fast',
  },
  {
    id: 'gemma-3',
    name: 'Gemma 3',
    description: 'Free open-source model by Google',
    isPremium: false,
    provider: 'google',
    category: 'fast',
  },
  {
    id: 'gemma-3n',
    name: 'Gemma 3n',
    description: 'Free compact model by Google',
    isPremium: false,
    provider: 'google',
    category: 'fast',
  },
  // Additional Gemini models (keeping them free as requested)
  {
    id: 'gemini-2.5-flash-preview',
    name: 'Gemini 2.5 Flash Preview',
    description: 'Latest preview model with enhanced capabilities',
    isPremium: false,
    provider: 'google',
    category: 'advanced',
  },
  {
    id: 'gemini-2.5-pro-preview',
    name: 'Gemini 2.5 Pro Preview',
    description: 'Most advanced Gemini model preview',
    isPremium: false,
    provider: 'google',
    category: 'advanced',
  },
  {
    id: 'gemini-2.0-flash',
    name: 'Gemini 2.0 Flash',
    description: 'Next-generation fast model with image generation',
    isPremium: false,
    provider: 'google',
    category: 'balanced',
  },
  {
    id: 'gemini-2.0-flash-lite',
    name: 'Gemini 2.0 Flash Lite',
    description: 'Lightweight version of Gemini 2.0 Flash',
    isPremium: false,
    provider: 'google',
    category: 'fast',
  },
  {
    id: 'gemini-1.5-flash',
    name: 'Gemini 1.5 Flash',
    description: 'Optimized for speed and efficiency',
    isPremium: false,
    provider: 'google',
    category: 'fast',
  },
  {
    id: 'gemini-1.5-flash-8b',
    name: 'Gemini 1.5 Flash 8B',
    description: 'Compact model for quick responses',
    isPremium: false,
    provider: 'google',
    category: 'fast',
  },
  {
    id: 'gemini-1.5-pro',
    name: 'Gemini 1.5 Pro',
    description: 'Enhanced Gemini with larger context',
    isPremium: false,
    provider: 'google',
    category: 'advanced',
  },
  // Additional Gemini models from official API
  {
    id: 'gemini-2.5-flash-preview-05-20',
    name: 'Gemini 2.5 Flash Preview (05-20)',
    description: 'Specific preview version of Gemini 2.5 Flash',
    isPremium: false,
    provider: 'google',
    category: 'advanced',
  },
  {
    id: 'gemini-2.5-pro-preview-06-05',
    name: 'Gemini 2.5 Pro Preview (06-05)',
    description: 'Specific preview version of Gemini 2.5 Pro',
    isPremium: false,
    provider: 'google',
    category: 'advanced',
  },
  {
    id: 'gemini-embedding-exp-03-07',
    name: 'Gemini Embedding Experimental',
    description: 'Experimental embedding model',
    isPremium: false,
    provider: 'google',
    category: 'fast',
  },
  {
    id: 'aqa',
    name: 'Attributed QA',
    description: 'Attributed Question-Answering model',
    isPremium: false,
    provider: 'google',
    category: 'balanced',
  },
];

// Convenience functions
export const getFreeModels = (): string[] => {
  return AI_MODELS.filter(model => !model.isPremium).map(model => model.id);
};

export const getPremiumModels = (): string[] => {
  return AI_MODELS.filter(model => model.isPremium).map(model => model.id);
};

export const getModelById = (id: string): ModelOption | undefined => {
  return AI_MODELS.find(model => model.id === id);
};

export const getModelsByProvider = (provider: string): ModelOption[] => {
  return AI_MODELS.filter(model => model.provider === provider);
};

export const getModelsByCategory = (category: string): ModelOption[] => {
  return AI_MODELS.filter(model => {
    if (Array.isArray(model.category)) {
      return model.category.includes(category as 'fast' | 'balanced' | 'advanced');
    }
    return model.category === category;
  });
};

export const isModelPremium = (modelId: string): boolean => {
  const model = getModelById(modelId);
  return model?.isPremium ?? false;
};

// Default models for different use cases
export const DEFAULT_MODELS = {
  FREE: 'gpt-3.5-turbo',
  PREMIUM: 'gpt-4o',
  FAST: 'gemini-flash',
  BALANCED: 'gemini-pro',
  ADVANCED: 'claude-3-sonnet',
} as const;

export type DefaultModelType = keyof typeof DEFAULT_MODELS;