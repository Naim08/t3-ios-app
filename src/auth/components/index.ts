export { AuthInput } from './AuthInput';
export { ModalHeader } from './ModalHeader';
export { Divider } from './Divider';
export { BaseAuthModal } from './BaseAuthModal';