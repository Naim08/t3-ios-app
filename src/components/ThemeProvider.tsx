import React, {
  createContext,
  useContext,
  useState,
  useEffect,
  ReactNode,
} from 'react';
import { Appearance } from 'react-native';
import { Theme, createTheme, ColorScheme } from '../theme';

interface ThemeContextType {
  theme: Theme;
  colorScheme: ColorScheme;
  toggleTheme: () => void;
  setColorScheme: (scheme: ColorScheme) => void;
}

const ThemeContext = createContext<ThemeContextType | undefined>(undefined);

interface ThemeProviderProps {
  children: ReactNode;
}

export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
  const [colorScheme, setColorScheme] = useState<ColorScheme>(
    (Appearance.getColorScheme() as ColorScheme) || 'light',
  );

  useEffect(() => {
    const subscription = Appearance.addChangeListener(({ colorScheme }) => {
      setColorScheme((colorScheme as ColorScheme) || 'light');
    });

    return () => subscription.remove();
  }, []);

  const theme = createTheme(colorScheme);

  const toggleTheme = () => {
    setColorScheme((prev) => (prev === 'light' ? 'dark' : 'light'));
  };

  return (
    <ThemeContext.Provider
      value={{ theme, colorScheme, toggleTheme, setColorScheme }}
    >
      {children}
    </ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const context = useContext(ThemeContext);
  if (!context) {
    throw new Error('useTheme must be used within a ThemeProvider');
  }
  return context;
};