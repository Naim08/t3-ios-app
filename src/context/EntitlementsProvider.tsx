import React, { createContext, useContext, useState, useMemo } from 'react';

export interface EntitlementsContextType {
  isSubscriber: boolean;
  hasCustomKey: boolean;
  remainingTokens: number;
  setSubscriber: (value: boolean) => void;
  setCustomKey: (value: boolean) => void;
  setRemainingTokens: (value: number) => void;
}

const EntitlementsContext = createContext<EntitlementsContextType | undefined>(undefined);

export const EntitlementsProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [isSubscriber, setSubscriber] = useState(false);
  const [hasCustomKey, setCustomKey] = useState(false);
  const [remainingTokens, setRemainingTokens] = useState(25000); // Mock: 25K tokens remaining

  const contextValue = useMemo(
    () => ({
      isSubscriber,
      hasCustomKey,
      remainingTokens,
      setSubscriber,
      setCustomKey,
      setRemainingTokens,
    }),
    [isSubscriber, hasCustomKey, remainingTokens]
  );

  return (
    <EntitlementsContext.Provider value={contextValue}>
      {children}
    </EntitlementsContext.Provider>
  );
};

export const useEntitlementsContext = () => {
  const context = useContext(EntitlementsContext);
  if (context === undefined) {
    throw new Error('useEntitlementsContext must be used within an EntitlementsProvider');
  }
  return context;
};
