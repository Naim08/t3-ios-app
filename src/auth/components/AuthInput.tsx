import React from 'react';
import { View, TextInput, TextInputProps } from 'react-native';
import { Mail, Lock, User } from 'lucide-react-native';
import { useTheme } from '../../components/ThemeProvider';
import { authModalStyles } from '../styles/authModalStyles';

interface AuthInputProps extends Omit<TextInputProps, 'style'> {
  icon: 'mail' | 'lock' | 'user';
  hasError?: boolean;
}

const getIconComponent = (icon: AuthInputProps['icon'], theme: any): React.ReactElement => {
  const iconProps = { 
    size: 20, 
    color: theme.colors.textSecondary, 
    strokeWidth: 2 
  };
  
  switch (icon) {
    case 'mail':
      return <Mail {...iconProps} />;
    case 'lock':
      return <Lock {...iconProps} />;
    case 'user':
      return <User {...iconProps} />;
    default:
      return <Mail {...iconProps} />;
  }
};

export const AuthInput: React.FC<AuthInputProps> = ({
  icon,
  hasError = false,
  ...textInputProps
}) => {
  const { theme } = useTheme();

  return (
    <View 
      style={[
        authModalStyles.inputWrapper, 
        { 
          borderColor: hasError 
            ? 'rgba(255, 107, 107, 0.6)' 
            : textInputProps.value 
              ? theme.colors.brand['400'] 
              : 'rgba(0, 0, 0, 0.1)' 
        }
      ]}
    >
      <View style={authModalStyles.inputIcon}>
        {getIconComponent(icon, theme)}
      </View>
      <TextInput
        style={[authModalStyles.input, { color: theme.colors.textPrimary }]}
        placeholderTextColor={theme.colors.textSecondary}
        autoCapitalize="none"
        autoCorrect={false}
        {...textInputProps}
      />
    </View>
  );
};