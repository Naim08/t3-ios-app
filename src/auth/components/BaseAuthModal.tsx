import React, { ReactNode } from 'react';
import {
  Modal,
  View,
  TouchableWithoutFeedback,
  KeyboardAvoidingView,
  Platform,
  ScrollView,
  Animated,
} from 'react-native';
import { useTheme } from '../../components/ThemeProvider';
import { useModalAnimation } from '../../hooks/useModalAnimation';
import { ModalHeader } from './ModalHeader';
import { authModalStyles } from '../styles/authModalStyles';

// Conditional import for BlurView
let BlurView: any;
try {
  const blurLib = require('expo-blur');
  BlurView = blurLib.BlurView;
} catch (error) {
  BlurView = View;
}

interface BaseAuthModalProps {
  visible: boolean;
  onClose: () => void;
  title: string;
  children: ReactNode;
  showDragIndicator?: boolean;
}

export const BaseAuthModal: React.FC<BaseAuthModalProps> = ({
  visible,
  onClose,
  title,
  children,
  showDragIndicator = true,
}) => {
  const { theme } = useTheme();
  const { slideAnim, fadeAnim, scaleAnim } = useModalAnimation({ visible });

  const handleBackdropPress = () => {
    onClose();
  };

  return (
    <Modal
      visible={visible}
      transparent
      animationType="none"
      statusBarTranslucent
      onRequestClose={onClose}
    >
      <KeyboardAvoidingView
        style={{ flex: 1 }}
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      >
        <View style={authModalStyles.overlay}>
          {/* Animated Backdrop */}
          <Animated.View
            style={[
              authModalStyles.backdrop,
              { opacity: fadeAnim }
            ]}
          >
            <TouchableWithoutFeedback onPress={handleBackdropPress}>
              {Platform.OS === 'ios' ? (
                <BlurView
                  intensity={80}
                  tint="dark"
                  style={authModalStyles.backdropTouch}
                />
              ) : (
                <View style={authModalStyles.backdropTouch} />
              )}
            </TouchableWithoutFeedback>
          </Animated.View>

          {/* Animated Modal Container */}
          <Animated.View
            style={[
              authModalStyles.container,
              {
                backgroundColor: theme.colors.surface,
                transform: [
                  { translateY: slideAnim },
                  { scale: scaleAnim }
                ],
              },
            ]}
          >
            <ScrollView
              style={{ flex: 1 }}
              contentContainerStyle={authModalStyles.scrollContent}
              keyboardShouldPersistTaps="handled"
              showsVerticalScrollIndicator={false}
            >
              <ModalHeader
                title={title}
                onClose={onClose}
                showDragIndicator={showDragIndicator}
              />
              
              {children}
            </ScrollView>
          </Animated.View>
        </View>
      </KeyboardAvoidingView>
    </Modal>
  );
};