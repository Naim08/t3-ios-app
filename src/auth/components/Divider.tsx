import React from 'react';
import { View, Text } from 'react-native';
import { useTheme } from '../../components/ThemeProvider';
import { authModalStyles } from '../styles/authModalStyles';

interface DividerProps {
  text?: string;
}

export const Divider: React.FC<DividerProps> = ({ text = 'OR' }) => {
  const { theme } = useTheme();

  return (
    <View style={authModalStyles.dividerContainer}>
      <View style={[authModalStyles.divider, { backgroundColor: theme.colors.border }]} />
      <Text style={[authModalStyles.dividerText, { color: theme.colors.textSecondary }]}>
        {text}
      </Text>
      <View style={[authModalStyles.divider, { backgroundColor: theme.colors.border }]} />
    </View>
  );
};