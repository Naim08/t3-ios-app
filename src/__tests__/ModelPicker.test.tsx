import React from 'react';
import { render } from '@testing-library/react-native';
import { ModelPickerSheet } from '../models/ModelPickerSheet';
import { EntitlementsProvider } from '../context/EntitlementsProvider';
import { ThemeProvider } from '../components/ThemeProvider';

// Mock react-i18next
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string, params?: any) => {
      if (key === 'models.remainingTokens' && params?.count) {
        return `${params.count} tokens remaining this month`;
      }
      return key;
    },
  }),
}));

// Mock bottom sheet
jest.mock('@gorhom/bottom-sheet', () => {
  const React = require('react');
  const BottomSheet = ({ children }: any) => {
    return (
      <div data-testid="bottom-sheet">
        {children}
      </div>
    );
  };
  
  BottomSheet.BottomSheetView = ({ children }: any) => {
    return <div data-testid="bottom-sheet-view">{children}</div>;
  };
  
  return BottomSheet;
});

// Simple test to generate snapshot
test('ModelPickerSheet renders correctly', () => {
  const mockProps = {
    isVisible: true,
    onClose: jest.fn(),
    onSelect: jest.fn(),
    onNavigateToPaywall: jest.fn(),
    remainingTokens: 25000,
  };

  const { toJSON } = render(
    <ThemeProvider>
      <EntitlementsProvider>
        <ModelPickerSheet {...mockProps} />
      </EntitlementsProvider>
    </ThemeProvider>
  );
  
  expect(toJSON()).toMatchSnapshot();
});
