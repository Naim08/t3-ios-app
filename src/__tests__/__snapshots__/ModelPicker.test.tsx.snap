// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelPickerSheet renders correctly 1`] = `
<div
  data-testid="bottom-sheet"
>
  <View
    style={
      [
        {
          "alignSelf": "center",
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingTop": 8,
          "width": "100%",
        },
        {
          "maxWidth": 600,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 24,
        }
      }
    >
      <Text
        accessibilityRole="text"
        allowFontScaling={true}
        maxFontSizeMultiplier={3}
        style={
          [
            {
              "includeFontPadding": false,
            },
            {
              "color": "#212529",
              "fontFamily": "System",
              "fontSize": 24,
              "fontWeight": "700",
              "lineHeight": 32,
              "textAlign": "left",
            },
          ]
        }
      >
        models.selectModel
      </Text>
      <Text
        accessibilityRole="text"
        allowFontScaling={true}
        maxFontSizeMultiplier={3}
        style={
          [
            {
              "includeFontPadding": false,
            },
            {
              "color": "#6C757D",
              "fontFamily": "System",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16,
              "marginTop": 4,
              "textAlign": "left",
            },
          ]
        }
      >
        25000 tokens remaining this month
      </Text>
    </View>
    <View
      style={
        {
          "gap": 12,
        }
      }
    >
      <View
        accessibilityHint="models.selectModelHint"
        accessibilityLabel="GPT-3.5: Fast and reliable for everyday tasks"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "minHeight": 44,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 12,
                "elevation": 2,
                "padding": undefined,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.15000000000000002,
                "shadowRadius": 4,
              },
              {
                "borderColor": "#CED4DA",
                "borderRadius": 12,
                "borderWidth": 1,
                "padding": 16,
                "position": "relative",
              },
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                }
              }
            >
              <Text
                accessibilityRole="text"
                allowFontScaling={true}
                maxFontSizeMultiplier={3}
                style={
                  [
                    {
                      "includeFontPadding": false,
                    },
                    {
                      "color": "#212529",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "lineHeight": 24,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                GPT-3.5
              </Text>
            </View>
            <Text
              accessibilityRole="text"
              allowFontScaling={true}
              maxFontSizeMultiplier={3}
              style={
                [
                  {
                    "includeFontPadding": false,
                  },
                  {
                    "color": "#6C757D",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "lineHeight": 18,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Fast and reliable for everyday tasks
            </Text>
          </View>
        </View>
      </View>
      <View
        accessibilityHint="models.unlockRequiredHint"
        accessibilityLabel="GPT-4o: Latest model with enhanced reasoning"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "minHeight": 44,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 12,
                "elevation": 2,
                "padding": undefined,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.15000000000000002,
                "shadowRadius": 4,
              },
              {
                "borderColor": "#CED4DA",
                "borderRadius": 12,
                "borderWidth": 1,
                "padding": 16,
                "position": "relative",
              },
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                }
              }
            >
              <Text
                accessibilityRole="text"
                allowFontScaling={true}
                maxFontSizeMultiplier={3}
                style={
                  [
                    {
                      "includeFontPadding": false,
                    },
                    {
                      "color": "#212529",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "lineHeight": 24,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                GPT-4o
              </Text>
            </View>
            <Text
              accessibilityRole="text"
              allowFontScaling={true}
              maxFontSizeMultiplier={3}
              style={
                [
                  {
                    "includeFontPadding": false,
                  },
                  {
                    "color": "#6C757D",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "lineHeight": 18,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Latest model with enhanced reasoning
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 12,
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "paddingHorizontal": 16,
                  "paddingVertical": 12,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "backgroundColor": "#FFFFFFE6",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 6,
                    "marginBottom": 8,
                    "paddingHorizontal": 8,
                    "paddingVertical": 4,
                  },
                  {
                    "backgroundColor": "#3970FF",
                  },
                ]
              }
            >
              <Text
                accessibilityRole="text"
                allowFontScaling={true}
                maxFontSizeMultiplier={3}
                style={
                  [
                    {
                      "includeFontPadding": false,
                    },
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontWeight": "600",
                      "lineHeight": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                models.premium
              </Text>
            </View>
            <View
              accessibilityHint="models.unlockHint"
              accessibilityLabel="models.unlockAccessibility"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#3970FF",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "minHeight": 44,
                  "minWidth": 120,
                  "opacity": 1,
                  "paddingHorizontal": 12,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                accessibilityRole="text"
                allowFontScaling={true}
                maxFontSizeMultiplier={3}
                style={
                  [
                    {
                      "includeFontPadding": false,
                    },
                    {
                      "color": "#3970FF",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "600",
                      "lineHeight": 20,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                models.unlock
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityHint="models.unlockRequiredHint"
        accessibilityLabel="Claude Sonnet: Advanced reasoning and analysis"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "minHeight": 44,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 12,
                "elevation": 2,
                "padding": undefined,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.15000000000000002,
                "shadowRadius": 4,
              },
              {
                "borderColor": "#CED4DA",
                "borderRadius": 12,
                "borderWidth": 1,
                "padding": 16,
                "position": "relative",
              },
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                }
              }
            >
              <Text
                accessibilityRole="text"
                allowFontScaling={true}
                maxFontSizeMultiplier={3}
                style={
                  [
                    {
                      "includeFontPadding": false,
                    },
                    {
                      "color": "#212529",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "lineHeight": 24,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Claude Sonnet
              </Text>
            </View>
            <Text
              accessibilityRole="text"
              allowFontScaling={true}
              maxFontSizeMultiplier={3}
              style={
                [
                  {
                    "includeFontPadding": false,
                  },
                  {
                    "color": "#6C757D",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "lineHeight": 18,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Advanced reasoning and analysis
            </Text>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "borderRadius": 12,
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "paddingHorizontal": 16,
                  "paddingVertical": 12,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "backgroundColor": "#FFFFFFE6",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "borderRadius": 6,
                    "marginBottom": 8,
                    "paddingHorizontal": 8,
                    "paddingVertical": 4,
                  },
                  {
                    "backgroundColor": "#3970FF",
                  },
                ]
              }
            >
              <Text
                accessibilityRole="text"
                allowFontScaling={true}
                maxFontSizeMultiplier={3}
                style={
                  [
                    {
                      "includeFontPadding": false,
                    },
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 12,
                      "fontWeight": "600",
                      "lineHeight": 16,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                models.premium
              </Text>
            </View>
            <View
              accessibilityHint="models.unlockHint"
              accessibilityLabel="models.unlockAccessibility"
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#3970FF",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "minHeight": 44,
                  "minWidth": 120,
                  "opacity": 1,
                  "paddingHorizontal": 12,
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                accessibilityRole="text"
                allowFontScaling={true}
                maxFontSizeMultiplier={3}
                style={
                  [
                    {
                      "includeFontPadding": false,
                    },
                    {
                      "color": "#3970FF",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "600",
                      "lineHeight": 20,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                models.unlock
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityHint="models.selectModelHint"
        accessibilityLabel="Gemini Pro: Google's advanced multimodal model"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 10,
            "left": 10,
            "right": 10,
            "top": 10,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "minHeight": 44,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 12,
                "elevation": 2,
                "padding": undefined,
                "shadowColor": "#000000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.15000000000000002,
                "shadowRadius": 4,
              },
              {
                "borderColor": "#CED4DA",
                "borderRadius": 12,
                "borderWidth": 1,
                "padding": 16,
                "position": "relative",
              },
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                }
              }
            >
              <Text
                accessibilityRole="text"
                allowFontScaling={true}
                maxFontSizeMultiplier={3}
                style={
                  [
                    {
                      "includeFontPadding": false,
                    },
                    {
                      "color": "#212529",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "600",
                      "lineHeight": 24,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Gemini Pro
              </Text>
            </View>
            <Text
              accessibilityRole="text"
              allowFontScaling={true}
              maxFontSizeMultiplier={3}
              style={
                [
                  {
                    "includeFontPadding": false,
                  },
                  {
                    "color": "#6C757D",
                    "fontFamily": "System",
                    "fontSize": 12,
                    "fontWeight": "400",
                    "lineHeight": 18,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Google's advanced multimodal model
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</div>
`;
